src/net.c

```diff
@@ -454,7 +473,11 @@ int valkeyContextConnectTcp(valkeyContext *c, const valkeyOptions *options) {
     }
     for (p = servinfo; p != NULL; p = p->ai_next) {
     addrretry:
-        if ((s = socket(p->ai_family, p->ai_socktype, p->ai_protocol)) == VALKEY_INVALID_FD)
+        rv = valkeyContextSetMptcp(c, p->ai_protocol);
+        if (rv == VALKEY_ERR)
+            goto error;
+
+        if ((s = socket(p->ai_family, p->ai_socktype, rv)) == VALKEY_INVALID_FD)
```

Comment:
Let's change this in the same way we changed it in the Valkey PR?

```suggestion
        rv = valkeyContextTcpGetProtocol(c->flags & VALKEY_MPTCP);
        if ((s = socket(p->ai_family, p->ai_socktype, rv)) == VALKEY_INVALID_FD)
```

Reply:
There is a config check for `IPPROTO_MPTCP` in valkey, so `valkeyContextTcpGetProtocol` will not fail.
sadly we don't have the same check in libvalkey.